File: leetcode/1614/maximum_nesting_depth_of_parentheses.cpp
#include <string>

using namespace std;

class Solution {

public:
  int maxDepth(string s) {
    int count = 0;
    int maxCount = 0;
    for (char c : s) {
      if (c == '(') {
        count++;
      } else if (c == ')') {
        count--;
      }

      if (count > maxCount) {
        maxCount = count;
      }
    }

    return maxCount;
  }
};

File: leetcode/1614/maximum_nesting_depth_of_parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        count = 0
        maxCount = 0
        for c in s:
            if c == '(':
                count += 1
            elif c == ')':
                count -= 1

            maxCount = max(count, maxCount)

        return maxCount
